File: app/src/main/java/edu/gatech/donatrix/model/ItemManager.java
package edu.gatech.donatrix.model;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class ItemManager {
    private final List<Item> items;

    /**
     * Constructor
     */
    public ItemManager() {
        this.items = new ArrayList<>();
    }

    //Paul: Shouldn't getItems be a method of Inventory?
    public List<Item> getItems() {
        return Collections.unmodifiableList(this.items);
    }

    //Paul: Shouldn't addItem be a method of Inventory?
    public void addItem(Item item) {
        this.items.add(item);
    }

    //Paul: Shouldn't removeITem be a method of Inventory?
    public void removeItem(Item item) {
        this.items.remove(item);
    }
}

File: app/src/main/java/edu/gatech/donatrix/model/ItemManager.kt
package edu.gatech.donatrix.model

import java.util.ArrayList
import java.util.Collections

internal class ItemManager {
    private val items: MutableList<Item>

    /**
     * Constructor
     */
    init {
        this.items = ArrayList()
    }

    //Paul: Shouldn't getItems be a method of Inventory?
    fun getItems(): List<Item> {
        return Collections.unmodifiableList(this.items)
    }

    //Paul: Shouldn't addItem be a method of Inventory?
    fun addItem(item: Item) {
        this.items.add(item)
    }

    //Paul: Shouldn't removeITem be a method of Inventory?
    fun removeItem(item: Item) {
        this.items.remove(item)
    }
}